packages/dds/tree/src/simple-tree/api/verboseTree.ts

```diff
@@ -0,0 +1,415 @@
+ */
+export type VerboseTree<THandle = IFluidHandle> =
+	| VerboseTreeNode<THandle>
+	| Exclude<TreeLeafValue, IFluidHandle>
+	| THandle;
+
+/**
+ * Verbose encoding of a {@link TreeNode}.
+ * @remarks
+ * This is verbose meaning that every {@link TreeNode} has an explicit `type` property, and `fields`.
+ * This allowed VerboseTreeNode to be unambiguous regarding which type each node is without relying on symbols or hidden state.
+ *
+ * Any IFluidHandle values have been replaced by `THandle`. If the `THandle` is JSON compatible, then this type is JSON compatible as well.
+ *
+ * @privateRemarks
+ * This type is only used for data which is copied into and out of the tree.
+ * When being copied out, its fine to have the data be mutable since its a copy.
+ *
+ * When being copied in, we don't need to mutate, so we could use a readonly variant of this type.
+ * however the copy in case (createFromVerbose) probably isn't harmed much by just reusing this type as is,
+ * since if the caller has immutable data, TypeScript doesn't prevent assigning immutable data to a mutable type anyway.
+ * Also relaxing the input methods to take readonly data would be a non-breaking change so it can be done later if desired.
+ *
+ * This format is simple-tree specialized alternative to {@link JsonableTree}.
+ * This format allows for all simple-tree compatible trees to be represented.
+ *
+ * Unlike `JsonableTree`, leaf nodes are not boxed into node objects, and instead have their schema inferred from the value.
+ * Additionally, sequence fields can only occur on a node that has a single sequence field (with the empty key) replicating the behavior of simple-tree ArrayNodes.
+ */
+export interface VerboseTreeNode<THandle = IFluidHandle> {
+	/**
+	 * The meaning of this node.
+	 * Provides contexts/semantics for this node and its content.
+	 * @remarks
+	 * Typically used to associate a node with metadata (including a schema) and source code (types, behaviors, etc).
+	 * When used with this package's schema system, it will be the {@link TreeNodeSchemaCore.identifier}.
+	 */
+	type: string;
+
+	/**
+	 * Content of this node.
+	 * For array nodes, an array of children.
+	 * For map and object nodes, an object which children under keys.
+	 * @remarks
+	 * For object nodes, the keys could be either the stored keys, or the property keys depending on usage.
+	 */
+	fields:
+		| VerboseTree<THandle>[]
+		| {
+				[key: string]: VerboseTree<THandle>;
+		  };
+}
+
+/**
+ * Options for how to interpret a `VerboseTree<TCustom>` when schema information is available.
+ */
+export interface ParseOptions<TCustom> {
+	/**
+	 * Fixup custom input formats.
+	 * @remarks
+	 * Main usage is to handle IFluidHandles.
+	 * When targeting JSON compatibility,
+	 * this may be by throwing an error or including a placeholder.
+	 * Since IFluidHandles are special references to FLuid data which is garbage collected when not referenced by the container for long enough,
```

Comment:
```suggestion
	 * Since IFluidHandles are special references to Fluid data which is garbage collected when not referenced by the container for long enough,
```